connection_configuration_test.go

```diff
@@ -0,0 +1,260 @@
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	err = os.Chmod("./test_data/snowflake/session/token", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "no-token-path")
+
+	cfg, err := LoadConnectionConfig()
+	assertNilF(t, err, "The error should not occur")
+	assertEqualF(t, cfg.Authenticator, AuthTypeOAuth)
+	assertEqualF(t, cfg.Token, "mock_token123456")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "read-token")
+
+	cfg, err = LoadConnectionConfig()
+	assertNilF(t, err, "The error should not occur")
+	assertEqualF(t, cfg.Authenticator, AuthTypeOAuth)
+	assertEqualF(t, cfg.Token, "mock_token123456")
+}
+
+func TestLoadConnectionConfigWitNonExistingDSN(t *testing.T) {
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "unavailableDSN")
+
+	_, err = LoadConnectionConfig()
+	assertNotNilF(t, err, "The error should occur")
+
+	driverErr, ok := err.(*SnowflakeError)
+	assertTrueF(t, ok, "This should be a Snowflake Error")
+	assertEqualF(t, driverErr.Number, ErrCodeFailedToFindDSNInToml)
+}
+
+func TestLoadConnectionConfigWithTokenFileNotExist(t *testing.T) {
+	err := os.Chmod("./test_data/connections.toml", 0600)
+	assertNilF(t, err, "The error occurred because you cannot change the file permission")
+
+	os.Setenv("SNOWFLAKE_HOME", "./test_data")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "aws-oauth-file")
+
+	_, err = LoadConnectionConfig()
+	assertNotNilF(t, err, "The error should occur")
+
+	_, ok := err.(*(fs.PathError))
+	assertTrueF(t, ok, "This error should be a path error")
+}
+
+func TestParseInt(t *testing.T) {
+	var i interface{}
+
+	i = 20
+	num, err := parseInt(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, num, 20)
+
+	i = "40"
+	num, err = parseInt(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, num, 40)
+
+	i = "wrong_num"
+	_, err = parseInt(i)
+	assertNotNilF(t, err, "should have failed")
+}
+
+func TestParseBool(t *testing.T) {
+	var i interface{}
+
+	i = true
+	b, err := parseBool(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, b, true)
+
+	i = "false"
+	b, err = parseBool(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, b, false)
+
+	i = "wrong_bool"
+	_, err = parseInt(i)
+	assertNotNilF(t, err, "should have failed")
+}
+
+func TestParseDuration(t *testing.T) {
+	var i interface{}
+
+	i = 300
+	dur, err := parseDuration(i)
+	assertNilF(t, err, "This value should be parsed")
+	assertEqualF(t, dur, time.Duration(5*int64(time.Minute)))
```

Comment:
I think using `time.Second` would be more readable

Reply:
Fixed.

Reply:
Done.